llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1750,6 +1774,8 @@ static bool eliminateConstraints(Function &F, DominatorTree &DT, LoopInfo &LI,
         Changed |= Simplified;
       } else if (auto *MinMax = dyn_cast<MinMaxIntrinsic>(Inst)) {
         Changed |= checkAndReplaceMinMax(MinMax, Info, ToRemove);
+      } else if (auto *CmpIntrinsic = dyn_cast<class CmpIntrinsic>(Inst)) {
```

Comment:
```suggestion
      } else if (auto *CmpIntrinsic = dyn_cast<CmpIntrinsic>(Inst)) {
```

Reply:
That doesn't work because the variable has the same name as the class, so the compiler assumes that the variable is used in its own initializer.

Reply:
Oh, I see. In that case please make it `auto *Cmp =` or something...

Reply:
Done